### PR TITLE
DATAP-1583 - company received date filter fixes

### PR DESCRIPTION
**Pull request by flacoman91**
fixing company received date filter

This filter was not clearing when you attempt to clear it, either by clicking Clear in the date picker, or clicking "Clear all filters"

## Additions

- added action to clear the filter

## Removals

- removed some convoluted logic where we were trying to be clever and reuse the same action for different filters

## Changes

-

## Testing

- yarn start
- Add dates to the "The date the CFPB sent the complaint to the company" filter
- see dates get applied

Screenshot 2024-11-01 at 2 44 42 PM

- Click Clear in each date to confirm the date can be cleared

Also Click "Clear all filters" to clear out the Company received date filter

## Notes

-

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/reducers/query/querySlice.js [hunk 1]
```diff
@@ -131,27 +148,24 @@ export const querySlice = createSlice({
           : null;
 
         const datesChanged =
-          state[fields[0]] !== minDate || state[fields[1]] !== maxDate;
+          state.date_received_min !== minDate ||
```
Comment (cdmh219) [inline src/reducers/query/querySlice.js:151]: is there a missing expression after the `||` clause, or is it line 152?

Comment (flacoman91) [reply src/reducers/query/querySlice.js:151]: it's line 152. 

the autoformatter puts it on the next line due to line length.

Comment (cdmh219) [reply src/reducers/query/querySlice.js:151]: Got it, thanks!

**Review by cdmh219 — APPROVED** [review]

**Comment by flacoman91** [comment]
Merged manually bc merge queue is hanging

**Pull request closed without merging**